src/server.js

```diff
@@ -46,20 +68,22 @@ app.get("/presentations/:id", (req, res) => {
   });
 });
 
-// temporary to hold favorite state in server state until we get viewstate in place
-let favorites = [];
-
 app.post("/search", (req, res) => {
+  const favorites = req.session.favorite_presentations;
```

Comment:
This fixes a bug where the favorite state is not respected when performing searches.